libglusterfs/src/client_t.c

```diff
@@ -459,47 +433,40 @@ client_ctx_set(client_t *client, void *key, void *value)
     return value;
 }
 
-int
-client_ctx_get(client_t *client, void *key, void **value)
+void *
+client_ctx_get(client_t *client, void *key)
 {
-    int ret = 0;
+    void *value = NULL;
 
     if (!client || !key)
-        return -1;
+        goto out;
 
     LOCK(&client->scratch_ctx_lock);
     {
-        ret = __client_ctx_get_int(client, key, value);
+        value = __client_ctx_get_int(client, key);
     }
     UNLOCK(&client->scratch_ctx_lock);
 
-    return ret;
+out:
+    return value;
 }
 
 static int
 __client_ctx_del_int(client_t *client, void *key, void **value)
 {
     int index = 0;
-    int ret = 0;
 
     for (index = 0; index < GF_CLIENTCTX_INITIAL_SIZE; index++) {
-        if (client->scratch_ctx[index].ctx_key == key)
-            break;
-    }
-
-    if (index == GF_CLIENTCTX_INITIAL_SIZE) {
-        ret = -1;
-        goto out;
+        if (client->scratch_ctx[index].ctx_key == key) {
+            if (value)
```

Comment:
Why we can't use the same approach here ? if this returns NULL, it means that the key wasn't present. Otherwise it will return the existing value.